src/ngraph/runtime/cpu/pass/cpu_memory_assignment.cpp

```diff
@@ -712,21 +712,16 @@ bool runtime::cpu::pass::CPUMemoryAssignment::run_on_function(shared_ptr<ngraph:
                             auto input_op = std::static_pointer_cast<op::Op>(input_node);
                             if (auto input_op_annotations = input_op->get_op_annotations())
                             {
-                                // when input is cacheable, do not allow destructive oi
-                                if (input_op_annotations->is_cacheable())
-                                {
-                                    NGRAPH_DEBUG << "cpu_memory_assignment: cacheable input, no "
-                                                    "destructive oi";
-                                    continue;
-                                }
                                 // when reusing memory, ops with different cacheabilities are using different memory manager
                                 // and should not share the same buffer.
-                                else if (!m_disable_memory_sharing &&
-                                         op_annotations->is_cacheable())
+                                if (!m_disable_memory_sharing &&
```

Comment:
just a trivial comment/question for later: should we rename the variable as `m_memory_shared_enabled` instead of using double negatives most of the times? `!` and `disable` ?

Reply:
`m_enable_memory_sharing` would be better, but it is a good idea.